### PR TITLE
`resource-path` should be taken into account for resource embeding 

### LINKED ISSUE #5084: `resource-path` not working as expected in HTML document
### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/5080

<div type='discussions-op-text'>

<sup>Originally posted by **sje30** April  4, 2023</sup>
I'm trying to use --resource-path to point to a different folder for where I store images.

With the following simple example:

````
---
title: "test 1"
resource-path: ["/tmp"]
format:
  html:
    embed-resources: true
---
# Generate a plot

```{r}
png(file="/tmp/temp.png")
plot(1:10)
dev.off()
```


# show the plot v1

![alt text](/tmp/temp.png "tooltip")

# show the plot v2

![alt text](temp.png "tooltip")
````

when I render in quarto, v1 of the plot appears, but not v2.

    quarto render temp.qmd

However, if I run pandoc from the command line, it works (as in the plot appears twice):

    pandoc --resource-path='/tmp' --self-contained -s -o temp.html temp.qmd

what am I missing?

Thanks, Stephen

```
pandoc v 3.1.1
quarto  v 1.2.475
running on Mac
```</div>

### PR DESCRIPTION
**Pull request by cderv**
For HTML, it seems we are doing ourself the `--embed-resource`. 

We need to take into account `resource-path` configuration in our process otherwise, resources won't be embeded correctly as with the initial feature.

@dragonstyle assigning you as reviewer - I believe this fix #5084. 

This could be postpone to 1.4 maybe if too close to 1.3 release. This usage is not popular I believe, but still not working without the PR.

### CODE REVIEW

**Comment by dragonstyle** [comment]
This looks like a good fix to me. I think we should discuss whether to take into 1.3 - it does seem like a safe fix since it targets only embedding...

File: src/core/pandoc/self-contained.ts [hunk 1]
```diff
@@ -33,6 +36,7 @@ export const pandocIngestSelfContainedContent = async (file: string) => {
   cmd.push("--output", filename);
   cmd.push("--metadata", "title=placeholder");
   cmd.push("--embed-resources");
+  resourcePath ? cmd.push("--resource-path", resourcePath.join(":")) : null;
```
Comment (cscheid) [inline src/core/pandoc/self-contained.ts:39]: Our style tends to be to use more explicit statement. Can we just use an if statement here?

Comment (cderv) [reply src/core/pandoc/self-contained.ts]: Oh sure. I'll remember that. Thanks ! 

> Our style tends to be to use more explicit statement. 

We don't have those rule written somewhere right ? By "our style" your are referring to most of current codebase style ? Meaning if I want to know the style to use, it is by observation of the rest of the code ? 

Just curious to improve my next PRs.

Comment (cscheid) [reply src/core/pandoc/self-contained.ts]: > Oh sure. I'll remember that. Thanks !
> 
> > Our style tends to be to use more explicit statement.
> 
> We don't have those rule written somewhere right ? By "our style" your are referring to most of current codebase style ? Meaning if I want to know the style to use, it is by observation of the rest of the code ?

Yes, that's what I mean.

But it's also what I think is _correct_ :). Specifically: putting a side-effect in the middle of an expression that you need to know is only conditionally evaluated is asking for trouble. You have `: null` there that isn't doing anything either. Your statement minimizes number of characters, but that's not really a great measure for readability. The replacement is explicit about conditional code paths, and doesn't need a dangling empty `else` clause.

**Comment by cscheid** [comment]
FWIW, I think we should take this change.

**Review by cscheid — APPROVED** [review]

File: src/core/pandoc/self-contained.ts (hunk 1 above)
Comment (cderv) [reply src/core/pandoc/self-contained.ts]: Understood. It is clear now. Thank you for the great review and feedback !

File: tests/docs/smoke-all/2023/04/04/issue-5084.qmd [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (allenmanning) [inline tests/docs/smoke-all/2023/04/04/issue-5084.qmd:1]: @cderv Is this testing the reveal code path as well?

Comment (cderv) [reply tests/docs/smoke-all/2023/04/04/issue-5084.qmd:1]: No only html format. We would need to add `revealjs` at least in `_quarto.yml` 

I'll add it. thanks

Comment (cderv) [reply tests/docs/smoke-all/2023/04/04/issue-5084.qmd:1]: Done in #5107

**Review by ghost — COMMENTED** [review]
LGTM

**Pull request merged**